spec/consensus/light-client.md

```diff
@@ -1,318 +1,494 @@
+hold more than 2/3 of the voting power in `validators(b.Header.NextValidatorsHash)` is correct until time
+`b.Header.Time + TRUSTED_PERIOD`.
+
+*Assumption*: "correct" is defined w.r.t. realtime (some Newtonian global notion of time, i.e., wall time),
+while `Header.Time` corresponds to the [BFT time](bft-time.md). In this note, we assume that clocks of correct processes
+are synchronized (for example using NTP), and therefore there is bounded clock drift between clocks and
+BFT time. We can make this more precise eventually (incorporating clock drift, accuracy, precision, etc.). Right now,
+we consider this assumption sufficient, as clock synchronization (under NTP) is in the order of milliseconds and
+`TRUSTED_PERIOD`  is in the order of weeks.
+
+*Remark*: This failure model might change to a hybrid version that takes heights into account in the future.
+
+The specification in this document considers an implementation of the lite client under the Tendermint Failure Model. Issues
+like `counter-factual signing`, `fork accountability` and `evidence submission` are mechanisms that justify this assumption by
+incentivizing validators to follow the protocol. If they don't, and we have 1/3 (or more) faults, safety may be violated.
+Our approach then is to *detect* these cases (after the fact), and take suitable repair actions (automatic and social).
+This is discussed in document on [Fork accountability](fork-accountability.md).
 
 ### Functions
 
-For the purpose of this lite client specification, we assume that the Tendermint Full Node exposes the following function over Tendermint RPC:
+**VerifyHeader.** The function `VerifyHeader` captures high level logic, i.e., application call to the lite client module to (optionally download) and
+verify header for some height. The core verification logic is captured by `CanTrust` function that iteratively try to establish trust in given header
+by relying on `CheckSupport` function.
 
 ```go
-    func Commit(height int64) (SignedHeader, error)
-      // returns signed header: header (with the fields from
-      // above) with Commit that include signatures of
-      // validators that signed the header
+func VerifyHeader(height, trustThreshold) error {
+  if untrusted_h, exists := Store.Get(height); exists {
+    if isWithinTrustedPeriod(untrusted_h) return nil
+    return ErrHeaderNotWithinTrustedPeriod(untrusted_h)
+  }
+
+  untrusted_h := Commit(height)
+  if !verify(untrusted_h) { return ErrInvalidHeader(untrusted_h) }
+  if !isWithinTrustedPeriod(untrusted_h) { return ErrHeaderNotWithinTrustedPeriod(untrusted_h) }
+
+  // get the highest trusted headers lower than untrusted_h
+  trusted_h = Store.HighestTrustedSmallerThan(height)
+  if trusted_h == nil { return ErrNoTrustedHeader }
+
+  err = CanTrust(trusted_h, untrusted_h, trustThreshold)  // or CanTrustBisection((trusted_h, untrusted_h, trustThreshold)
+  if err != nil { return err }
+
+  if isWithinTrustedPeriod(untrusted_h) {
```

Comment:
If this re-checking is necessary, I think it needs an explanation (see https://github.com/tendermint/spec/pull/61#discussion_r361158266).